recipes/stlab/all/conanfile.py

```diff
@@ -1,180 +1,177 @@
-
-        if self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) < "15.8":
-            raise ConanInvalidConfiguration("Need Visual Studio >= 2017 15.8 (MSVC 19.15)")
-
-        # Actually, we want *at least* 15.8 (MSVC 19.15), but we cannot check this for now with Conan.
-        if self.settings.compiler == "msvc" and Version(self.settings.compiler.version) < "19.15":
-            raise ConanInvalidConfiguration("Need msvc >= 19.15")
+            check_min_cppstd(self, self._minimum_cpp_standard)
 
+        self._validate_min_compiler_version()
         self._validate_task_system()
+        self._validate_thread_system()
         self._validate_boost_components()
 
     def configure(self):
-        if self.options.task_system == "auto":
-            self.options.task_system = self._default_task_system()
-        self.output.info("Stlab Task System: {}.".format(self.options.task_system))
+        self.output.info("STLab With Boost: {}.".format(self.options.with_boost))
+        self.output.info("STLab Future Coroutines: {}.".format(self.options.future_coroutines))
+        self.output.info("STLab No Standard Coroutines: {}.".format(self.options.no_std_coroutines))
+        self.output.info("STLab Task System: {}.".format(self.options.task_system))
+        self.output.info("STLab Thread System: {}.".format(self.options.thread_system))
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables['BUILD_TESTING'] = not self.conf.get("tools.build:skip_test", default=True, check_type=bool)
+        tc.variables["STLAB_USE_BOOST_CPP17_SHIMS"] = self.options.with_boost
+        tc.variables["STLAB_NO_STD_COROUTINES"] = self.options.no_std_coroutines
+        tc.variables["STLAB_THREAD_SYSTEM"] = self.options.thread_system
+        tc.variables["STLAB_TASK_SYSTEM"] = self.options.task_system
+        tc.generate()
+
+        tc = CMakeDeps(self)
+        tc.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
 
     def package(self):
-        self.copy("*LICENSE", dst="licenses", keep_path=False)
-        self.copy("stlab/*", src=self._source_subfolder, dst='include/')
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
+        cmake.install()
+
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rm(self, "msvcp*.dll", os.path.join(self.package_folder, "bin"))
+        rm(self, "concrt*.dll", os.path.join(self.package_folder, "bin"))
+        rm(self, "vcruntime*.dll", os.path.join(self.package_folder, "bin"))
 
     def package_id(self):
-        self.info.header_only()
-        self.info.options.boost_optional = "ANY"
-        self.info.options.boost_variant = "ANY"
+        # TODO: is header only but needs a header modified by cmake
+        # self.info.settings.clear()
+        # self.info.header_only()
+        pass
```

Comment:
@fpelliccioni Sorry, could you please give me more information about this one? why can't you put `self.info.settings.clear()`?